### DISCUSSION TITLE
rust and numeric computation

### DISCUSSION THREAD

**Discussion by jimy-byerley** [rust-ndarray/ndarray#1364] [Ideas]
Hello there
I am a random engineer using ndarray for numeric computation.

I just [posted on the rust internals forum](https://internals.rust-lang.org/t/rust-and-numeric-computation/20425) about missing features in rust to allow numeric computation code (and crates) better.
I would be interested in having the opinion of the ndarray community - especially the ndarray developper - about it :)

**Comment by nilgoyette** [comment]
I don't have much to contribute except "I agree". 

I'm a "small" maintainer here, I'm mostly answering question and merging small PRs. I know ndarray as a programmer, not as a maintainer. In brief, I don't know much about the internals. I do know about the `Ix#` "hack". If ndarray was coded today, it would probably use something else!

Because writing generic functions in Rust is "ugly", I tend to choose a float type and stick with it. Later, I realize that I also need the other float type so I make it generic over `num_traits` types. Slightly related, I wrote [this function](https://github.com/Enet4/nifti-rs/blob/master/src/writer.rs#L417) for another project that I maintain. It's ... not sexy. I don't really want to be writing that kind of code. I was a C++ programmer a while ago and I did like the freedom to write generic code in that langage. (I still prefer Rust)
